### PR TITLE
Business logic

### PR DESCRIPTION
**Pull request by isomethane**

### CODE REVIEW

File: internal/interface/httpapi/api.go [hunk 1]
```diff
@@ -73,36 +72,176 @@ type getMusicalEntityResponseModel struct {
...
+			w.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		em := getMusicalEntityResponseModel{
+			Artist: e.Artist,
+			Album: e.Album,
+			Track: e.Track,
+			Links: make([]link, 0, len(e.Links)),
+		}
+		for _, l := range em.Links {
```
Comment (M0nsteR) [inline internal/interface/httpapi/api.go]: Could you explain me the reason behind iterating the empty Links slice here?

File: internal/interface/httpapi/api.go [hunk 2]
```diff
@@ -73,36 +72,176 @@ type getMusicalEntityResponseModel struct {
...
+			em.Links = append(em.Links, link{
```
Comment (M0nsteR) [inline internal/interface/httpapi/api.go]: Doesn't look nice to extend the same array you're iterating. Even though it works in Go, the coder's intent is not clear.

File: internal/interface/httpapi/api.go [hunk 3]
```diff
@@ -73,36 +72,176 @@ type getMusicalEntityResponseModel struct {
...
+	p, err := a.PlaylistUseCases.GetPlaylistById(pid)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	m := getPlaylistResponseModel{
+		Name: p.Name,
+		Content: make([]getMusicalEntityResponseModel, 0, len(p.Content)),
+	}
+	for _, c := range p.Content {
```
Comment (M0nsteR) [inline internal/interface/httpapi/api.go:115]: It's time to introduce a function here: too many nested loops are always dangerous.

**Review by M0nsteR — CHANGES_REQUESTED** [review]
Please improve the readability of your code.

File: internal/interface/httpapi/api.go (hunk 1 above)
Comment (isomethane) [reply internal/interface/httpapi/api.go]: Fixed.

File: internal/interface/httpapi/api.go (hunk 2 above)
Comment (isomethane) [reply internal/interface/httpapi/api.go]: Fixed.

**Review by M0nsteR — APPROVED** [review]
Good to go.

**Pull request merged**